Tests/MutationSteps/BuildForTestingTests.swift

```diff
@@ -137,6 +137,26 @@ final class BuildForTestingTests: MuterTestCase {
             .literal(reason: "Could not parse xctestrun at path: some/project.xctestrun")
         )
     }
+    
+    func test_givenTestArgumentsContainsDerivedDataPath_shouldCopyXCTestrunCorrectly() async throws {
+        state.muterConfiguration = MuterConfiguration(
+            executable: "/path/to/xcodebuild",
+            arguments: [
+                "some",
+                "commands",
+                "-derivedDataPath",
+                "/path/to/custom_derived_Data_path",
+                "test"
+            ]
+        )
+        
+        state.mutatedProjectDirectoryURL = URL(fileURLWithPath: "/path/to/temp")
+        process.stdoutToBeReturned = xcodebuildBuildForTestingOutput()
+        
+        _ = try? await sut.run(with: state)
```

Comment:
You should be able to do this, because the test function is marked `throws`, unless you expect it to actually throw an error while testing? In that case, I'd probably prefer `XCTAssertThrows` or an explicit do/catch or something.

```suggestion
        _ = try await sut.run(with: state)
```

Reply:
I only provided just enough mock data to make the `run` func execute the `copy xctestrun` logic, so the `run` function will throw eventually.
Since I only care about the `copy` logic get executed, so I used `try?`